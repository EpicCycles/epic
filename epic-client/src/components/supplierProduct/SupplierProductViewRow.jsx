import React, {Fragment} from "react";
import * as PropTypes from "prop-types";
import {getComponentKey} from "../app/model/helpers/model";
import {supplierProductFields} from "../app/model/helpers/fields";
import ViewModelField from "../app/model/ViewModelField";


const SupplierProductViewRow = (props) => {
    const { supplierProduct, lockFirstColumn, suppliers } = props;
    const componentKey = getComponentKey(supplierProduct);
    return <Fragment>
        {supplierProductFields.map((field, index) => {
            const fixed = ((index === 0) && lockFirstColumn) ? "grid-item--fixed-left" : "";
            return <div
                className={`grid-item ${fixed}`}
                key={`SupplierProductRow${field.fieldName}${componentKey}`}
            >
                <ViewModelField
                    field={field}
                    model={supplierProduct}
                    suppliers={suppliers}
                />
            </div>
        })}
    </Fragment>
};
SupplierProductViewRow.propTypes = {
    supplierProduct: PropTypes.object.isRequired,
    lockFirstColumn: PropTypes.bool,
    suppliers: PropTypes.array.isRequired,
};

export default SupplierProductViewRow;
