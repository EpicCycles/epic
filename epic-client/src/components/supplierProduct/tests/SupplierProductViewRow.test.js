import React from 'react';
import SupplierProductViewRow from "../SupplierProductViewRow";

const suppliers = [
    { id: 12, name: "supplier Name" },
    { id: 121, name: "supplier Name 2" },
];
test('should display when passed an empty supplier product', () => {
    const component = shallow(<SupplierProductViewRow supplierProduct={{}} suppliers={suppliers}/>);
    expect(component).toMatchSnapshot();
});
test('should display when passed supplier product', () => {
    const supplierProduct = {
        id: 121,
        supplier: 12,
        part: 3,
        product_code: 'PC101',
        fitted_price: 23.99,
        ticket_price: 24.99,
        rrp: 24.99,
        trade_price: 11,
        club_price: 22.99,
        check_date: new Date(),
    };
    const component = shallow(<SupplierProductViewRow supplierProduct={supplierProduct} suppliers={suppliers}/>);
    expect(component).toMatchSnapshot();
});
