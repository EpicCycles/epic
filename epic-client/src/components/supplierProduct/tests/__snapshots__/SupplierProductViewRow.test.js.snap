// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display when passed an empty supplier product 1`] = `
<Fragment>
  <div
    className="grid-item "
    key="SupplierProductRowsuppliernew"
  >
    <ViewModelField
      field={
        Object {
          "error": "A supplier must be selected",
          "fieldName": "supplier",
          "header": "Supplier",
          "required": true,
          "synonyms": Array [
            "supplier",
          ],
          "type": "supplier",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowpartnew"
  >
    <ViewModelField
      field={
        Object {
          "error": "A part must be selected",
          "fieldName": "part",
          "header": "Part",
          "required": true,
          "synonyms": Array [
            "part",
          ],
          "type": "part",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowproduct_codenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "product_code",
          "header": "Product Code",
          "length": 30,
          "synonyms": Array [
            "product_code",
          ],
          "type": "text",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowfitted_pricenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "fitted_price",
          "header": "Fitted Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowticket_pricenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "ticket_price",
          "header": "Ticket Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowrrpnew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "rrp",
          "header": "RRP",
          "length": 10,
          "synonyms": Array [
            "price",
            "selling price",
            "srp",
            "rrp",
            "sell price",
            "retail price",
          ],
          "type": "currency",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowtrade_pricenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "trade_price",
          "header": "Trade Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowclub_pricenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "club_price",
          "header": "Club Price",
          "length": 10,
          "type": "currency",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowcheck_datenew"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "check_date",
          "header": "Date Checked",
          "readonly": true,
          "type": "date_time",
        }
      }
      model={Object {}}
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
</Fragment>
`;

exports[`should display when passed supplier product 1`] = `
<Fragment>
  <div
    className="grid-item "
    key="SupplierProductRowsupplier121"
  >
    <ViewModelField
      field={
        Object {
          "error": "A supplier must be selected",
          "fieldName": "supplier",
          "header": "Supplier",
          "required": true,
          "synonyms": Array [
            "supplier",
          ],
          "type": "supplier",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowpart121"
  >
    <ViewModelField
      field={
        Object {
          "error": "A part must be selected",
          "fieldName": "part",
          "header": "Part",
          "required": true,
          "synonyms": Array [
            "part",
          ],
          "type": "part",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowproduct_code121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "product_code",
          "header": "Product Code",
          "length": 30,
          "synonyms": Array [
            "product_code",
          ],
          "type": "text",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowfitted_price121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "fitted_price",
          "header": "Fitted Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowticket_price121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "ticket_price",
          "header": "Ticket Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowrrp121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "rrp",
          "header": "RRP",
          "length": 10,
          "synonyms": Array [
            "price",
            "selling price",
            "srp",
            "rrp",
            "sell price",
            "retail price",
          ],
          "type": "currency",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowtrade_price121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "trade_price",
          "header": "Trade Price",
          "length": 10,
          "synonyms": Array [],
          "type": "currency",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowclub_price121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "club_price",
          "header": "Club Price",
          "length": 10,
          "type": "currency",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
  <div
    className="grid-item "
    key="SupplierProductRowcheck_date121"
  >
    <ViewModelField
      field={
        Object {
          "fieldName": "check_date",
          "header": "Date Checked",
          "readonly": true,
          "type": "date_time",
        }
      }
      model={
        Object {
          "check_date": 2015-03-25T18:30:00.000Z,
          "club_price": 22.99,
          "fitted_price": 23.99,
          "id": 121,
          "part": 3,
          "product_code": "PC101",
          "rrp": 24.99,
          "supplier": 12,
          "ticket_price": 24.99,
          "trade_price": 11,
        }
      }
      suppliers={
        Array [
          Object {
            "id": 12,
            "name": "supplier Name",
          },
          Object {
            "id": 121,
            "name": "supplier Name 2",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
